src/metabase/models/card.clj

```diff
@@ -767,6 +767,112 @@
+
+(defn- update-mapping
+  "Return modifed mapping, or nil, according to action."
+  [identifier->action mapping]
+  (if-not (and (= :dimension (get-in mapping [:target 0]))
+               (#{:field :expression} (get-in mapping [:target 1 0])))
+    mapping
+    (let [dimension (get-in mapping [:target 1])
+          identifier (subvec dimension 0 2)
+          [action arg] (get identifier->action identifier [:noop])]
+      (case action
+        :update (assoc-in mapping [:target 1] arg)
+        :noop   mapping
+        :delete nil))))
+
+(defn- update-for-dashcard
+  "Return nil for no-op updates or [<dashcard-id> <update>]. <update> is a map of `:parameter_mappings` only as only
+  this is updated."
+  [identifier->action dashcard]
+  (let [card-update (select-keys (update dashcard :parameter_mappings
+                                         (comp vec (partial keep (partial update-mapping identifier->action))))
+                                 [:parameter_mappings])]
+    (when (not= (:parameter_mappings dashcard) (:parameter_mappings card-update))
+      [(:id dashcard) card-update])))
+
+(defn- updates-for-dashcards
+  [identifier->action dashcards]
+  (not-empty (keep (partial update-for-dashcard identifier->action) dashcards)))
+
+;; ! :delete action is shut down until I find reasonable condition for deletion.
+(defn- update-associated-parameters!
+  "Update _parameter mappings_ of _dashcards_ that target modified _card_, to reflect the modification.
+
+  This function handles only modifications to breakout.
+
+  Context. Card can have multiple multiple breakout elements referencing same field or expression, having different
+  _temporal unit_. Those refs can be targeted by dashboard _temporal unit parameter_. If refs change, and card is saved,
+  _parameter mappings_ have to be updated to target new, modified refs. This function takes care of that.
+
+  First mappings of _identifier_ -> _action_ are generated. _identifier_ is described
+  eg. in [[breakouts-->identifier->action]] docstring. Then, dashcards are fetched and updates are generated
+  by [[updates-for-dashcards]]. Updates are then executed."
+  [card-before card-after]
+  (let [card->breakout     #(-> % :dataset_query mbql.normalize/normalize :query :breakout)
+        breakout-before    (card->breakout card-before)
+        breakout-after     (card->breakout card-after)]
+    (when-some [identifier->action (breakouts-->identifier->action breakout-before breakout-after)]
+      (let [dashcards          (t2/select :model/DashboardCard :card_id (some :id [card-after card-before]))
+            updates            (updates-for-dashcards identifier->action dashcards)]
+        ;; Beware. This can have negative impact on card update performance as queries are fired in sequence. I'm not
+        ;; aware of more reasonable way.
+        (doseq [[id update] updates]
+          (t2/update! :model/DashboardCard :id id update))))))
```

Comment:
I'm unaware of better way. Please let me know if you do.

Reply:
Probably it's better to do these in one transaction, to keep the index locking etc. under control. Otherwise I'm not aware of a better way to do this.

Reply:
Thanks. Done.